docs/language_guide.md

```diff
@@ -0,0 +1,619 @@
+
+### Data
+You will have various kinds of data passing through your systems. Sysl allows you to express ownership, information classification and other attributes of your data in one place.
+
+Continuing with the previous example, let's define a `Server` that expects `LoginData` for the `Login` Flow.
+```
+Server:
+  !type LoginData
+    username <: string
+    password <: string
+```
+Note: we have not defined how `Server` implements the `Login` flow.
+
+### Projects
+Most of the changes to your system will be done as part of a well defined `project` or a `software release`.
+
+`TODO: Elaborate`
+## Generate Diagrams
+
+Sysl supports generating diagrams of following types:
+  * Sequence Diagram
+  * Integration Diagram
+  * Data Diagram
+
+Sysl aims to generate code and documentation from only one source of truth i.e. `.sysl` files.
+
+## Generate Code
+
+A common problem with documentation is that the code and documentation diverge pretty soon. `Sysl` lets to define `Rest API` and generate code for the same. You can describe the behaviour of the api for developers to follow and generate code and diagrams from same `.sysl` code. See section on [Rest](#rest) below.
+
+Currently, Sysl generates Java code for the following
+  * Sprint REST Controller
+  * POJO classes for the types used in your api
+  * Transformation Language to transform your data models e.g. from relational to object model (and vice-versa?)
+
+## Parts of Sysl Language
+Let's see how we can define application and data models in sysl.
+
+### Application, Endpoint
+As described above, here is how you can start to define an application.
+
+`mobile.sysl`
+```
+MobileApp:
+  Login: ...
+  Search: ...
+  Order: ...
+```
+
+Sysl expects all lines after `:` to be indented. The only exception to this rule is when you want to use the shortcut `...`. Sysl has support to generate code for defining `Rest` api servers. See [Rest](#rest) for more details.
+
+### Data Models
+#### Data Types
+Sysl supports following data types out of the box.
+  * int, int64, int32
+  * float, decimal
+  * string
+  * bool
+  * datetime, date
+  * any
+  * xml
+
+#### Relational Data Model
+Relational Data model is the most common way of persisting data in a database. You can define your data model directly in sysl.
+`data.sysl`
+```java
+CustomerOrderModel:
+  !table Customer:
+    customer_id <: int [~pk]
+
+  !table CustomerOrder:
+    order_id <: int [~pk]
+    customer_id <: Customer.customer_id
+```
+In the above example:
+  *  Customer.customer_id is a primary key.
+  *  CustomerOrder has a foreign key customer_id which refers to the primary key of Customer (i.e. customer_id).
+
+See [data.sysl](assets/data.sysl) for complete example for Relation model.
+
+![](assets/data-Relational-Model.png)
+
+#### Object Model
+Define a typical in memory Object model of an application
+```java
+CustomerOrderModel:
```

Comment:
i'm still confused about the syntax. is this top level `CustomerOrderModel` also an "application"?